doc/spring-boot-plugin-dev.md

```diff
@@ -0,0 +1,318 @@
+# Spring Boot Plugin Development
+
+This document explores the use cases and methodology for implementing,
+packaging and integrating Spring Boot applications which are extensible by external plugins.
+This document refers to application and library code written in Kotlin and managed by Maven,
+but should be applicable to any other equivalent technologies.
+
+When developing a Spring Boot application,
+we may want to import dependencies into the project which serve a specific set of features or integrations with other services,
+but are not necessarily useful to all deployments of the application.
+For example, an application which publishes or consumes files may - in some cases - need to integrate
+with an external service such as Amazon S3 via the Amazon SDK library,
+while in other cases local file storage may be sufficient.
+
+It is sometimes preferable to simply include those dependencies in the core application JAR, and make the related features optional.
+In this case, the application must support enabling and disabling those features
+through the local environment (environment variables, application.properties file, etc.) of each deployment of the application.
+
+Alternatively, we may prefer to include such features and their dependencies selectively as "plugins", based on the specific needs of each deployment.
+This is the focus of this document.
+This approach prevents the core application JAR and configuration files from being inflated with unnecessary dependencies.
+For features which utilise Spring auto-configuration, this approach also prevents the framework from attempting to
+configure beans and resources that are not needed and might emit irrelevant errors.
+
+By packaging optional features as plugins rather than building them into the core application,
+the application developer empowers the operator to decide which features and dependencies should be included in each deployment at the point of configuration.
+Furthermore, once we adapt the core application to support plugins using the method described in this document,
+developers can extend its functionality arbitrarily with any number of additional plugins, as long as they are compatible.
+
+## Overview
```

Comment:
This is very well writen. It might be biast because I have experience with Spring, but after reading it I know exactly what this article is about and what to expect